docs/examples/example.ipynb

```diff
@@ -335,32 +333,87 @@
+    "        result = [x for x in content[2].split(\" \") if x != \"\"]\n",
+    "        \n",
+    "        # MULTIPLY THE READINGS WITH MULTIPLIER\n",
+    "        wifi_signal = []\n",
+    "        for i in range(2:5):\n",
+    "            wifi_signal.append(int(result[i]) * self.multiplier)\n",
+    "        return {\"link\": wifi_signal[0], \"level\": wifi_signal[1], \"noise\": wifi_signal[2]}\n",
+    "```\n",
+    "\n",
+    "If a validator function is added, it must also be called when registering the resource. The `ResourceCreatorRegistration` should now also have the validator function passed in as the second parameter. \n",
+    "\n",
+    "```python\n",
+    "        async def main():\n",
+    "            Registry.register_resource_creator(Sensor.SUBTYPE, MySensor.MODEL, ResourceCreatorRegistration(MySensor.new, MySensor.validate_config))\n",
+    "```\n",
+    "\n",
+    "Reconfiguration specifies what values can change based on a new configuration. To allow this, add a `reconfigure()` function to the `MySensor` class.\n",
+    "\n",
+    "```python\n",
+    "        def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]):\n",
+    "            self.multiplier = config.attributes.fields[\"multiplier\"].number_value\n",
+    "```\n",
+    "\n",
+    "Click [here](https://gist.github.com/purplenicole730/dc3ddd9d53fd5da16c3bde0e0a6a63f4#file-module_step2_optional-py) to see the new `wifi_sensor.py` with the optional validator and reconfiguration functions."
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "\n",
     "### 3. Create an entry point file\n",
-    "The module now has to be created and started. In the `wifi_sensor.py` file, update the main function to the following. In a more complex module, it makes sense to create an entirely new entrypoint file. See [here](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/complex_module/src/main.py) for an example.\n",
+    "The module now has to be created and started. In the `wifi_sensor.py` file, update the main function to the following. If there is a validator function in the class, don't forget to add the function to the `ResourceCreatorRegistration` object. In a more complex module, it makes sense to create an entirely new entrypoint file. See [here](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/complex_module/src/main.py) for an example.\n",
     "\n",
     "```python\n",
     "# wifi-sensor/src/wifi_sensor.py\n",
-    "import asyncio\n",
-    "\n",
     "from viam.module.module import Module\n",
-    "from viam.components.sensor import Sensor\n",
-    "from viam.resource.registry import Registry, ResourceCreatorRegistration\n",
     "\n",
     "\n",
     "async def main():\n",
-    "    \"\"\"This function creates and starts a new module, after adding all desired resource model.\n",
+    "    \"\"\"\n",
```

Comment:
1. I can edit the title to something like "Start module".
2. If the starting of the module is not the entry point, then how was it approved before? This was not decided with this PR, and there have been multiple PRs since the first one, so I'm wondering how it was passed this entire time.
3. Adding `main()` was not part of this PR, so I don't know what drove this change. I will rename the file.